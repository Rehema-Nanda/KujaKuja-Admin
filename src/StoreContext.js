// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
