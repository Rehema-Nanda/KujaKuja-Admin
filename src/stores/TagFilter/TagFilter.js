import {
    decorate, computed, comparer,
} from "mobx";

class TagFilter {
    store = null;
    id = null;
    tagText = null;
    searchText = null;
    startDate = null;
    endDate = null;
    createdAt = null;
    updatedAt = null;
    status = null;
    locations = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            tag_text: this.tagText,
            search_text: this.searchText,
            start_date: this.startDate,
            end_date: this.endDate,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            status: this.status,
            locations: this.locations,
        };
    }

    updateFromJson(json) {
        this.tagText = json.tag_text;
        this.searchText = json.search_text;
        this.startDate = json.start_date;
        this.endDate = json.end_date;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.status = json.status;
        this.locations = json.locations;
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateTagFilter(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createTagFilter(json);
            this.id = parseInt(responseData.id, 10);
            this.status = this.status || "EDITING";
            this.createdAt = this.createdAt || new Date();
            await this.store.createTagFilter(this);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }

        await this.store.apiService.deleteTagFilter(this.id);
        await this.store.removeTagFilter(this);
    }
}

decorate(TagFilter, {
    asJson: computed({ equals: comparer.shallow }),
});

export default TagFilter;
