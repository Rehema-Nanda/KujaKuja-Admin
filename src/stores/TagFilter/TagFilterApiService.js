import ApiService from "../ApiService";

export default class TagFilterApiService extends ApiService {
    fetchAllTagFilters = async () => {
        const tagFiltersResponse = await this.makeHttpCall("tags/tag_filters");
        return tagFiltersResponse.data.data; // TODO: change data structure returned by API
    };

    runAllTagFilters = async () => {
        const tagFilters = await this.makeHttpCall("tags/bulk_tag");
        return tagFilters.data.data;
    };

    runTagFilterById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/bulk_tag/${id}`);
        return tagResponse.data.data;
    };

    updateTagFilter = async (id, tagJson) => {
        const config = { data: tagJson };
        const updateTagFilterResponse = await this.makeHttpCall(`tags/tag_filters/${id}`, "put", config);
        return updateTagFilterResponse.data;
    };

    createTagFilter = async (tagJson) => {
        const config = { data: tagJson };
        const createTagFilterResponse = await this.makeHttpCall(`tags/tag_filters`, "post", config);
        createTagFilterResponse.data.id = createTagFilterResponse.data.id.toString();
        return createTagFilterResponse.data;
    }

    deleteTagFilter = async (id) => {
        const deleteTagFilterResponse = await this.makeHttpCall(`tags/tag_filters/${id}`, "delete");
        deleteTagFilterResponse.data.id = deleteTagFilterResponse.data.id.toString();
        return deleteTagFilterResponse.data;
    };

    fetchTagFilterById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/tag_filters/${id}`);
        return tagResponse.data.data; // TODO: change data structure returned by API
    };

    undoTagFilterBulkTagById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/bulk_tag/undo/${id}`);
        return tagResponse.data.data;
    }
}
