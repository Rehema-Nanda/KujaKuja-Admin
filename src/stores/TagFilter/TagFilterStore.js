import {
    flow, decorate, observable, computed, onBecomeObserved,
} from "mobx";
import openSocket from "socket.io-client";
import TagFilter from "./TagFilter";
import TagFilterApiService from "./TagFilterApiService";
import AppConfig from "../../AppConfig";

class TagFilterStore {
    tagFilters = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new TagFilterApiService(AppConfig.API_URL);

        this.loadTagFilters();

        const socket = openSocket(AppConfig.ROOT_API_URL, { transports: ["polling"] });

        socket.on("connected", () => {
            socket.emit("listening for tag filter change", {});
        });

        const openStream = () => {
            socket.on("tagFilterTableChanged", () => {
                this.loadTagFilters();
            });
        };
        onBecomeObserved(this, "tagFilters", openStream);
    }

    get tagFiltersSortedByTagText() {
        return this.tagFilters.sort((a, b) => {
            if (a.tagText < b.tagText) {
                return -1;
            }
            else if (a.tagText > b.tagText) {
                return 1;
            }
            return 0;
        });
    }

    loadTagFilters = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;

        try {
            const tagFiltersJson = yield this.apiService.fetchAllTagFilters();

            this.tagFilters.replace(tagFiltersJson.map((tagFilterJson) => {
                return new TagFilter(
                    this,
                    tagFilterJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`TagFilterStore failed to load tags. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateTagFromServer = flow(function* (id) { // eslint-disable-line func-names
        const tagFilterId = parseInt(id, 10);
        this.isLoading = true;
        const tagsJson = yield this.apiService.fetchTagFilterById(id);
        let tagFilter = this.tagFilters.find((t) => {
            return t.id === tagFilterId;
        });

        if (!tagsJson && tagFilter) {
            // tag no longer exists on the server, remove it locally
            yield this.removeTagFilter(tagFilter);
            tagFilter = null;
        }
        if (tagsJson) {
            if (tagFilter) {
                tagFilter.updateFromJson(tagsJson);
            }
            else {
                tagFilter = new TagFilter(this, tagsJson);
                this.tagFilters.push(tagFilter);
            }
        }
        this.isLoading = false;
        return tagFilter;
    });

    removeTagFilter = flow(function* (tagFilter) { // eslint-disable-line func-names, require-yield
        this.tagFilters.splice(this.tagFilters.indexOf(tagFilter), 1);
    });

    createTagFilter = flow(function* (tagFilter) { // eslint-disable-line func-names, require-yield
        this.tagFilters.push(tagFilter);
    });
}

decorate(TagFilterStore, {
    tagFilters: observable,
    tagFiltersSortedByTagText: computed,
    isLoading: observable,
});

export default TagFilterStore;
