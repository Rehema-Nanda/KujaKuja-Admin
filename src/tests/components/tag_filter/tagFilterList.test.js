import React from "react";
import { mount } from "enzyme";
import TagFilterList from "../../../components/tag_filters/list";
import { StoreProvider } from "../../../StoreContext";
import TagFilterStore from "../../../stores/TagFilter/TagFilterStore";
import TagFilter from "../../../stores/TagFilter/TagFilter";

jest.mock("../../../stores/TagFilter/TagFilterStore");

TagFilterStore.mockImplementation(() => {
    return {
        tagFilters: [
            new TagFilter(
                null,
                {
                    tag_text: "testTag",
                    search_text: "test|sample|jest",
                    start_date: "2020-01-01 00:00:00",
                    end_date: "2020-01-09 23:59:59",
                    created_at: "2020-01-05 12:13:14",
                },
            ),
        ],
    };
});

describe("TagFilterList component", () => {
    const props = {
        history: {
            push: jest.fn(),
        },
    };

    const wrapper = mount(<TagFilterList {...props} />, {
        wrappingComponent: StoreProvider,
    });

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    xit("should display tag filter table", () => {
        expect(wrapper.find("table")).toHaveLength(1);
        expect(wrapper.find("th").at(3).text()).toBe("Results: 1");
    });

    it("should push history when add button is clicked", () => {
        const addElement = jest.spyOn(wrapper.instance(), "addElement");

        addElement();
        expect(props.history.push).toHaveBeenCalledWith("/tag_filters/add");
    });

    xit("should display a zero count on the tag filter table", () => {
        TagFilterStore.mockImplementationOnce(() => {
            return {
                tagFilters: [],
            };
        });

        const wrapper2 = mount(<TagFilterList />, {
            wrappingComponent: StoreProvider,
        });
        expect(wrapper2.find("th").at(3).text()).toBe("Results: 0");
    });
});
