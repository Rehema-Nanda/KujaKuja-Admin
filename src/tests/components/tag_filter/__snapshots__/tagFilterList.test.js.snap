// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagFilterList component renders correctly 1`] = `
"<TagFilterList history={{...}}>
  <div className=\\"page\\">
    <div className=\\"title-background-parent\\">
      <div className=\\"title-background-shape\\" />
    </div>
    <Container className=\\"desktop-non-fluid\\" tag=\\"div\\">
      <div className=\\"desktop-non-fluid container\\">
        <Row tag=\\"div\\">
          <div className=\\"row\\">
            <Col tag=\\"div\\" widths={{...}}>
              <div className=\\"col\\">
                <h1 className=\\"left\\">
                  Tag Filters
                </h1>
                <p>
                  Manage your Tag Filters
                </p>
              </div>
            </Col>
            <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
              <div className=\\"col-sm-5 col-md-2\\">
                <div className=\\"bulk-update-button\\">
                  <Dropdown isOpen={false} toggle={[Function]} direction=\\"down\\" nav={false} active={false} addonType={false} inNavbar={false} setActiveFromChild={false}>
                    <Manager tag=\\"div\\" className=\\"dropdown\\" onKeyDown={[Function: bound handleKeyDown]}>
                      <div className=\\"dropdown\\" onKeyDown={[Function: bound handleKeyDown]}>
                        <DropdownToggle caret={true} color=\\"primary\\" className=\\"primary\\" aria-haspopup={true}>
                          <Target aria-haspopup={true} color=\\"primary\\" cssModule={[undefined]} className=\\"primary dropdown-toggle\\" component={[Function: Button]} onClick={[Function: bound onClick]} aria-expanded={false}>
                            <Button aria-haspopup={true} color=\\"primary\\" cssModule={[undefined]} className=\\"primary dropdown-toggle\\" onClick={[Function: bound onClick]} aria-expanded={false} innerRef={[Function: targetRef]} tag=\\"button\\">
                              <button type=\\"button\\" aria-haspopup={true} onClick={[Function: bound onClick]} aria-expanded={false} className=\\"primary dropdown-toggle btn btn-primary\\" aria-label={{...}}>
                                Actions
                              </button>
                            </Button>
                          </Target>
                        </DropdownToggle>
                        <DropdownMenu tag=\\"div\\" flip={true}>
                          <div tabIndex=\\"-1\\" role=\\"menu\\" aria-hidden={true} className=\\"dropdown-menu\\" x-placement={[undefined]}>
                            <DropdownItem className=\\"bulk-update-dropdown\\" onClick={[Function]} tag=\\"button\\" toggle={true}>
                              <button type=\\"button\\" onClick={[Function: bound onClick]} tabIndex=\\"0\\" className=\\"bulk-update-dropdown dropdown-item\\">
                                <span className=\\"bulk-update-dropdown-title\\">
                                  Add filter
                                </span>
                              </button>
                            </DropdownItem>
                            <DropdownItem className=\\"bulk-update-dropdown\\" onClick={[Function]} tag=\\"button\\" toggle={true}>
                              <button type=\\"button\\" onClick={[Function: bound onClick]} tabIndex=\\"0\\" className=\\"bulk-update-dropdown dropdown-item\\">
                                <span className=\\"bulk-update-dropdown-title\\">
                                  Apply all filters
                                </span>
                              </button>
                            </DropdownItem>
                          </div>
                        </DropdownMenu>
                      </div>
                    </Manager>
                  </Dropdown>
                </div>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <div className=\\"white-page-bg\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col tag=\\"div\\" widths={{...}}>
                <div className=\\"col\\">
                  <div>
                    <BootstrapTableContainer bootstrap4={true} size=\\"sm\\" responsive=\\"sm\\" keyField=\\"id\\" data={{...}} columns={{...}} defaultSorted={{...}} pagination={{...}} striped={true} hover={true} bordered={false} noDataIndication=\\"No Data\\">
                      <DataProvider keyField=\\"id\\" columns={{...}} data={{...}}>
                        <SortProvider keyField=\\"id\\" columns={{...}} defaultSorted={{...}} defaultSortDirection={[undefined]} sort={[undefined]} data={{...}}>
                          <PaginationDataProvider pagination={{...}} data={{...}} bootstrap4={true} isRemotePagination={[Function]} remoteEmitter={{...}} onDataSizeChange={[undefined]}>
                            <BootstrapTable bootstrap4={true} size=\\"sm\\" responsive=\\"sm\\" keyField=\\"id\\" data={{...}} columns={{...}} defaultSorted={{...}} pagination={{...}} striped={true} hover={true} bordered={false} noDataIndication=\\"No Data\\" sortOrder=\\"asc\\" onSort={[Function]} sortField={{...}} setRemoteEmitter={[undefined]} remote={false} condensed={false} selectRow={{...}} expandRow={{...}} cellEdit={{...}} filterPosition=\\"inline\\">
                              <div className=\\"react-bootstrap-table\\">
                                <table id={[undefined]} className=\\"table table-striped table-hover\\">
                                  <Header columns={{...}} className={[undefined]} wrapperClasses={[undefined]} sortField={{...}} sortOrder=\\"asc\\" onSort={[Function]} globalSortCaret={[undefined]} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} selectRow={{...}} expandRow={{...}} filterPosition=\\"inline\\">
                                    <thead className={[undefined]}>
                                      <tr className={[undefined]}>
                                        <HeaderCell index={0} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                          <th tabIndex={0} aria-label=\\"Tag sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                            Tag
                                            <SortSymbol>
                                              <span className=\\"order-4\\" />
                                            </SortSymbol>
                                          </th>
                                        </HeaderCell>
                                        <HeaderCell index={1} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                          <th tabIndex={0} aria-label=\\"Search Text sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                            Search Text
                                            <SortSymbol>
                                              <span className=\\"order-4\\" />
                                            </SortSymbol>
                                          </th>
                                        </HeaderCell>
                                        <HeaderCell index={2} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                          <th tabIndex={0} aria-label=\\"Status sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                            Status
                                            <SortSymbol>
                                              <span className=\\"order-4\\" />
                                            </SortSymbol>
                                          </th>
                                        </HeaderCell>
                                        <HeaderCell index={3} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                          <th tabIndex={0} aria-label=\\"Created At sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                            Created At
                                            <SortSymbol>
                                              <span className=\\"order-4\\" />
                                            </SortSymbol>
                                          </th>
                                        </HeaderCell>
                                        <HeaderCell index={4} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                          <th tabIndex={0} className=\\"column-right\\">
                                            Results: 1
                                          </th>
                                        </HeaderCell>
                                      </tr>
                                    </thead>
                                  </Header>
                                  <Body className={[undefined]} data={{...}} keyField=\\"id\\" tabIndexCell={[undefined]} columns={{...}} isEmpty={false} visibleColumnSize={5} noDataIndication=\\"No Data\\" cellEdit={{...}} selectRow={{...}} expandRow={{...}} rowStyle={[undefined]} rowClasses={[undefined]} rowEvents={[undefined]}>
                                    <tbody className={[undefined]}>
                                      <SimpleRow row={{...}} tabIndexCell={[undefined]} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value={[undefined]} rowIndex={0} visibleColumnSize={5} attrs={{...}} style={{...}} className={{...}} editable={true}>
                                        <tr style={{...}} className={{...}}>
                                          <RowPureContent shouldUpdate={false} tabIndexStart={-1} row={{...}} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value={[undefined]} rowIndex={0} editable={true}>
                                            <Cell row={{...}} editable={true} rowIndex={0} columnIndex={0} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                              <td>
                                                <small>
                                                  <strong>
                                                    #
                                                    testTag
                                                  </strong>
                                                </small>
                                              </td>
                                            </Cell>
                                            <Cell row={{...}} editable={true} rowIndex={0} columnIndex={1} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                              <td>
                                                <small>
                                                  test|sample|jest
                                                </small>
                                              </td>
                                            </Cell>
                                            <Cell row={{...}} editable={true} rowIndex={0} columnIndex={2} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                              <td />
                                            </Cell>
                                            <Cell row={{...}} editable={true} rowIndex={0} columnIndex={3} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                              <td>
                                                05-Jan-2020 12:13
                                              </td>
                                            </Cell>
                                            <Cell row={{...}} editable={true} rowIndex={0} columnIndex={4} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]} className=\\"column-right\\">
                                              <td className=\\"column-right\\">
                                                <Button color=\\"primary\\" className=\\"primary\\" onClick={[Function: onClick]} tag=\\"button\\">
                                                  <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary btn btn-primary\\" aria-label={{...}}>
                                                    View
                                                  </button>
                                                </Button>
                                              </td>
                                            </Cell>
                                          </RowPureContent>
                                        </tr>
                                      </SimpleRow>
                                    </tbody>
                                  </Body>
                                </table>
                              </div>
                            </BootstrapTable>
                            <PaginationHandler sizePerPageList={{...}} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} dataSize={1} currPage={1} currSizePerPage={25}>
                              <Pagination sizePerPageList={{...}} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} paginationSize={5} showTotal={false} pageListRenderer={{...}} pageButtonRenderer={{...}} sizePerPageRenderer={{...}} paginationTotalRenderer={{...}} sizePerPageOptionRenderer={{...}} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} dataSize={1} currPage={1} currSizePerPage={25} lastPage={1} totalPages={1}>
                                <div className=\\"row react-bootstrap-table-pagination\\">
                                  <div className=\\"col-md-6 col-xs-6 col-sm-6 col-lg-6\\">
                                    <SizePerPageDropdownAdapter sizePerPageList={{...}} currSizePerPage={25} hideSizePerPage={false} sizePerPageRenderer={{...}} sizePerPageOptionRenderer={{...}} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                      <SizePerPageDropDown sizePerPageList={{...}} currSizePerPage=\\"25\\" hideSizePerPage={false} sizePerPageRenderer={{...}} sizePerPageOptionRenderer={{...}} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} options={{...}} optionRenderer={{...}} onClick={[Function: bound toggleDropDown]} onBlur={[Function: bound closeDropDown]} open={false} hidden={false} btnContextual=\\"btn-default btn-secondary\\" variation=\\"dropdown\\" className=\\"\\">
                                        <span style={{...}} className=\\"react-bs-table-sizePerPage-dropdown dropdown\\">
                                          <button id=\\"pageDropDown\\" className=\\"btn btn-default btn-secondary dropdown-toggle\\" data-toggle=\\"dropdown\\" aria-expanded={false} onClick={[Function: bound toggleDropDown]} onBlur={[Function: bound closeDropDown]}>
                                            25
                                             
                                          </button>
                                          <ul className=\\"dropdown-menu \\" role=\\"menu\\" aria-labelledby=\\"pageDropDown\\">
                                            <SizePerPageOption text=\\"25\\" page={25} bootstrap4={true} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" className=\\"dropdown-item\\" data-page={25} onMouseDown={[Function: onMouseDown]}>
                                                25
                                              </a>
                                            </SizePerPageOption>
                                            <SizePerPageOption text=\\"50\\" page={50} bootstrap4={true} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" className=\\"dropdown-item\\" data-page={50} onMouseDown={[Function: onMouseDown]}>
                                                50
                                              </a>
                                            </SizePerPageOption>
                                            <SizePerPageOption text=\\"100\\" page={100} bootstrap4={true} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" className=\\"dropdown-item\\" data-page={100} onMouseDown={[Function: onMouseDown]}>
                                                100
                                              </a>
                                            </SizePerPageOption>
                                            <SizePerPageOption text=\\"ALL\\" page={1} bootstrap4={true} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" className=\\"dropdown-item\\" data-page={1} onMouseDown={[Function: onMouseDown]}>
                                                ALL
                                              </a>
                                            </SizePerPageOption>
                                          </ul>
                                        </span>
                                      </SizePerPageDropDown>
                                    </SizePerPageDropdownAdapter>
                                  </div>
                                  <div className=\\"react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6\\">
                                    <PaginationListAdapter alwaysShowAllBtns={false} withFirstAndLast={true} paginationSize={5} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" currPage={1} currSizePerPage={25} pageStartIndex={1} lastPage={1} totalPages={1} pageButtonRenderer={{...}} onPageChange={[Function: bound handleChangePage]}>
                                      <PaginatonList pageButtonRenderer={{...}} onPageChange={[Function: bound handleChangePage]} pages={{...}}>
                                        <ul className=\\"pagination react-bootstrap-table-page-btns-ul\\">
                                          <PageButton page={1} active={true} disabled={false} title=\\"1\\" onPageChange={[Function: bound handleChangePage]}>
                                            <li className=\\"active page-item\\" title=\\"1\\">
                                              <a href=\\"#\\" onClick={[Function: bound handleClick]} className=\\"page-link\\">
                                                1
                                              </a>
                                            </li>
                                          </PageButton>
                                        </ul>
                                      </PaginatonList>
                                    </PaginationListAdapter>
                                  </div>
                                </div>
                              </Pagination>
                            </PaginationHandler>
                          </PaginationDataProvider>
                        </SortProvider>
                      </DataProvider>
                    </BootstrapTableContainer>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Container>
  </div>
</TagFilterList>"
`;
